test/cache_self_test.rb

```diff
@@ -80,6 +80,25 @@ def test_update
     profile.update_attributes(point: 10)
   end
 
+  def test_touch
+    time = Time.now
+    difficulty = Difficulty.create(updated_at: time)
+    assert_queries(1){ assert_equal time.to_i, Difficulty.cacher.find_by(id: difficulty.id).updated_at.to_i }
+    assert_queries(0){ assert_equal time.to_i, Difficulty.cacher.find_by(id: difficulty.id).updated_at.to_i }
+    assert_cache("active_model_cachers_Difficulty_#{difficulty.id}" => difficulty)
+
+    Time.stub :now, Time.at(0) do
+      assert_queries(1){ difficulty.touch }
```

Comment:
Currently, if you comment out [this line](https://github.com/khiav223577/active_model_cachers/pull/34/files#diff-20561ac5ac99ee3c42879c201d638f96R101), it'll still pass this test case.

Because the touch will trigger `after_commit` callback with wrong `previous_changes`, which I think it is a bug of `Rails`, see https://github.com/rails/rails/issues/32962.

Using `Difficulty.find(difficulty.id).touch` here is better. It can test if `after_touch` works fine or not.